.github/workflows/codeql-analysis.yml

```diff
@@ -14,11 +14,17 @@ on:
   schedule:
     - cron: '0 13 * * 6'
 
+permissions: {}
+
 jobs:
   analyze:
     name: Analyze
     runs-on: ubuntu-latest
 
+    permissions:
+      # See: https://github.com/github/codeql-action/blob/008b2cc71c4cf3401f45919d8eede44a65b4a322/README.md#usage
+      security-events: write
+
     steps:
```

Comment:
Keeping this permissioned scoped at the job-level even though there's only one job to mitigate impact of adding a new job in this workflow without proper evaluation of the GItHub token permissions.